src/error.jl

```diff
@@ -27,16 +25,62 @@ end
 function gdaljl_errorhandler(class::CPLErr, errno::Cint, errmsg::Cstring)
     # function signature needs to match the one in __init__, and the signature
     # of the callback for a custom error handler in the GDAL docs
-    if class in throw_class
+    if class === CE_Failure
         throw(GDALError(class, errno, unsafe_string(errmsg)))
     end
     return C_NULL
 end
 
-"Throw an error if a pointer is null and GDAL reports an error"
-function failsafe(ptr)
-    if ptr == C_NULL && cplgetlasterrortype() in throw_class
+"Check the last error type and throw a GDALError if it is a failure"
+function maybe_throw()
+    if cplgetlasterrortype() === CE_Failure
         throw(GDALError())
     end
-    ptr
+    nothing
+end
+
+"""
+Handle anything returned from GDAL
+
+When values are returned from GDAL, always check if there was a failure using `maybe_throw`.
+
+Depending on the return type, we do extra work.
+"""
+function aftercare(x)
```

Comment:
The naming can be more descriptive here. `gdal_errorcheck_and_free` is ugly, but more readable.

Reply:
Yeah, though that still wouldn't describe everything it does. Then it would have to be `gdal_errorcheck_and_free_and_load_strings` ;)
And if we add a new functionality, we have to rename the function. That is why I went for the less descriptive but more general `aftercare`, since we basically do anything we need to returned values, including error checking. One benefit of this is that if we want to tweak it, we just modify the `aftercare` definitions, and don't have to run the wrapper again every time.